viewer/lib/viewer.js

```diff
@@ -96,6 +100,53 @@ class Viewer {
     })
   }
 
+  updateTimecycleLighting (timeOfDay, moonPhase) {
+    if (timeOfDay === undefined) { return }
+    const lightIntensity = this.calculateIntensity(timeOfDay)
+    const newSkyColor = `#${this.darkenSkyColour(lightIntensity).padStart(6, 0)}`
+
+    function timeToRads (time) {
+      return time * (Math.PI / 12000)
+    }
+
+    // Update colours
+    this.scene.background = new THREE.Color(newSkyColor)
+    const newAmbientIntensity = Math.min(0.43, lightIntensity * 0.75) + (0.04 - (moonPhase / 100))
+    const newDirectionalIntensity = Math.min(0.63, lightIntensity) + (0.06 - (moonPhase / 100))
+    this.ambientLight.itensity = newAmbientIntensity
```

Comment:
Hey! I was interested in wether it could improve the day cycle. When I was trying to port it to TS i noticed it gives a type error because it should be `intensity` instead of `itensity` (a typo). However I'm still not sure of using intensity like this w setting position like this, it still looks not accurate (on the left side is the original client with minimal brightness and this code on the right):
<img width="600" src="https://github.com/PrismarineJS/prismarine-viewer/assets/46503702/01f44213-b23f-4e9e-a037-7451ffbb26d5">


At the very least you should not use intervals since time can be updated at any moment (night skip, time command), also isRaining can affect the sky color:
[commit](https://github.com/zardoy/prismarine-web-client/commit/0f9d4bc8946f6689aeee2f451bc259cccea5d058)

Reply:
Any improvements are very welcome - I will do a bit more testing but wont be able to do much about specific changes for weather lighting etc.

Reply:
As I remember weather don't affect lighting much - sky gets darker a bit and that's it (and imo it's enough). Anyway, it's too dark at almost any time. Also if you set evening you can see like native client doesn't really change directional light positioning. Really recommend running both clients connected to the same server so it's easy to compare

Reply:
this should be fixed

itensity -> intensity at least